src/main/java/com/icthh/xm/uaa/service/UserService.java

```diff
@@ -154,14 +154,14 @@ public User createUser(UserDTO user) {
     public Optional<UserDTO> updateUser(UserDTO updatedUser) {
         return userRepository.findById(updatedUser.getId())
             .map(user -> {
+                //use blockUserAccount/activateUserAccount
+                user.setActivated(user.isActivated());
+                //use changeUserRole
+                user.setRoleKey(user.getRoleKey());
```

Comment:
do we really need to set the value to the same value? maybe we can omit these two calls:
user.setActivated(user.isActivated());
user.setRoleKey(user.getRoleKey());

Reply:
it is **explict** set, to show we are saving same values, of course it can be removed